libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2416,7 +2416,7 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 	return m_functionCollector.createFunction(functionName, [&]() {
 		if (_from.baseType()->isValueType())
 		{
-			solAssert(_from.baseType() == _to.baseType(), "");
+			solAssert(*_from.baseType() == *_to.baseType(), "");
```

Comment:
Can this assert be reproduced outside of using user defined value type?

Reply:
Maybe for enums, but I didn't check. My change only relaxed the check, so I think it's fine.